x-pack/plugins/fleet/public/applications/fleet/sections/epm/screens/detail/index.test.tsx

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createTestRendererMock, MockedFleetStartServices, TestRenderer } from '../../../../mock';
+import { Detail } from './index';
+import React from 'react';
+import { PAGE_ROUTING_PATHS, pagePathGetters } from '../../../../constants';
+import { Route } from 'react-router-dom';
+import { GetInfoResponse } from '../../../../../../../common/types/rest_spec';
+import { DetailViewPanelName, KibanaAssetType } from '../../../../../../../common/types/models';
+
+describe('when on integration detail', () => {
+  let testRenderer: TestRenderer;
+  let renderResult: ReturnType<typeof testRenderer.render>;
+  const render = () =>
+    (renderResult = testRenderer.render(
+      <Route path={PAGE_ROUTING_PATHS.integration_details}>
+        <Detail />
+      </Route>
+    ));
+
+  beforeEach(() => {
+    testRenderer = createTestRendererMock();
+    mockApiCalls(testRenderer.startServices.http);
+    testRenderer.history.push(`${pagePathGetters.integration_details({ pkgkey: 'nginx-0.3.7' })}`);
+  });
+
+  describe('and a custom UI extension is NOT registered', () => {
+    beforeEach(() => render());
+
+    it('should show overview and settings tabs', () => {
+      const tabs: DetailViewPanelName[] = ['overview', 'settings'];
+      for (const tab of tabs) {
+        expect(renderResult.getByTestId(`tab-${tab}`));
+      }
+    });
+
+    it('should not show a custom tab', () => {
```

Comment:
I plan to complete these, but might not be part of this PR. I would like to get this due to the major surgery done the Plugin refactoring due to continued master merge conflicts.